scipy/stats/tests/test_morestats.py

```diff
@@ -545,7 +545,7 @@ def test_alternative_exact(self):
         # mass under the calculated statistic.
         astart, dist, _ = gscale(len(x1), len(x2))
         ind = int(statistic - astart)
-        prob = dist[ind] / np.sum(dist)
+        prob = np.float64(dist[ind]) / np.sum(dist)
```

Comment:
Does it make a difference here to convert the `dist[ind]` to `np.float64`?

Reply:
Because the test was written with an equality statement, and because `dist` is `float32` (whereas i upcast `freq` to float64 in the code), this was failing. 

So you could see that behavior wasn't changing, i left this direct call to `gscale`, but this should change to a call to the `pmf` of an `_ABW` instance.

Reply:
> Because the test was written with an equality statement,

Oh, yes. That was a mistake. I should have used `assert_allclose`.

> but this should change to a call to the `pmf` of an `_ABW` instance.

I agree. I will change it once this gets merged.